src/Microsoft.ML.Data/DataLoadSave/FakeSchema.cs

```diff
@@ -6,51 +6,52 @@
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Internal.Utilities;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace Microsoft.ML.Data.DataLoadSave
 {
-
     /// <summary>
     /// A fake schema that is manufactured out of a SchemaShape.
     /// It will pretend that all vector sizes are equal to 10, all key value counts are equal to 10,
     /// and all values are defaults (for metadata).
     /// </summary>
-    internal sealed class FakeSchema : ISchema
+    internal sealed class FakeSchemaFactory
```

Comment:
>FakeSchemaFactory [](start = 26, length = 17)

So @wschin, could we even just do away with most of this code altogether, and just have this be a static class with a single static method? I'm not sure what purpose creating this object actually serves, since you don't actually do anything with the instance per se. So why keep the type, when we can just nuke it?

Reply:
+1 to move it as a static method somewhere else

---
In reply to: [241493306](https://github.com/dotnet/machinelearning/pull/1872#discussion_r241493306) [](ancestors = 241493306)

Reply:
It's a fully-static class now. On the other hand, I don't see another utility function creating `Schema` out of `SchemaShape`, so let's keep it for now.

---
In reply to: [241493306](https://github.com/dotnet/machinelearning/pull/1872#discussion_r241493306) [](ancestors = 241493306)

Reply:
Yes -- though creating a `SchemaShape` out of a `Schema` is a generally useful utility, the utility of making "fake" schemas out of `SchemaShape` is somewhat limited, and I think probably something that will eventually go away as we convert things away from the old `IDataLoader` and `IDataTransform` idioms.